apps/omg_child_chain/lib/omg_child_chain/gas_price/poisson_gas_strategy.ex

```diff
@@ -0,0 +1,148 @@
+# Copyright 2020 OmiseGO Pte Ltd
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+defmodule OMG.ChildChain.GasPrice.PoissonGasStrategy do
+  @moduledoc """
+  Suggests gas prices based on Poisson regression model (also used by EthGasStation).
+
+  Ported from https://github.com/ethgasstation/gasstation-express-oracle/blob/master/gasExpress.py
+  """
+  use GenServer
+  require Logger
+
+  alias OMG.ChildChain.GasPrice
+  alias OMG.ChildChain.GasPrice.Strategy
+
+  @behaviour Strategy
+
+  @recalculate_interval_ms 60_000
+
+  @type t() :: %__MODULE__{
+          gas_price_to_use: pos_integer(),
+          max_gas_price: pos_integer()
+        }
+
+  defstruct gas_price_to_use: 20_000_000_000,
+            max_gas_price: 20_000_000_000
+
+  @doc """
+  Starts the EthGasStation strategy.
+  """
+  def start_link(args) do
+    GenServer.start_link(__MODULE__, args, name: __MODULE__)
+  end
+
+  @doc false
+  @impl GenServer
+  def init(args) do
+    state = %__MODULE__{
+      max_gas_price: Keyword.fetch!(args, :max_gas_price)
+    }
+
+    {:ok, state, {:continue, :start_recalculate}}
+  end
+
+  @doc false
+  @impl GenServer
+  def handle_continue(:start_recalculate, state) do
+    _ = send(self(), :recalculate)
+    {:ok, _} = :timer.send_interval(@recalculate_interval_ms, self(), :recalculate)
```

Comment:
Was going to use the [`handle_info(:timeout, ...` approach](https://hexdocs.pm/elixir/GenServer.html#module-timeouts) that @mederic-p shared with me, but this genserver will be accepting `:get_price` call regularly and so I think it will interfere with the timeout approach.